sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/SymmetricHashJoinStateManager.scala

```diff
@@ -497,4 +608,20 @@ object SymmetricHashJoinStateManager {
   private def getStateStoreName(joinSide: JoinSide, storeType: StateStoreType): String = {
     s"$joinSide-$storeType"
   }
+
+  /**
+   * Helper class for representing data key to (value, matched).
+   * Designed for object reuse.
+   */
+  case class KeyToValueAndMatched(
```

Comment:
I commented on the other class name, but the comment also applies to this one. `AndMatched` is very specific, I'd prefer to keep a more generic name (e.g. `KeyToValuePair` to kinda follow the old `UnsafeRowPair` name).